### PR TITLE
[ticket/14670] Use symfony 3.x in master branch

### PR DESCRIPTION
**Pull request by marc1706**
Checklist:

- [x] Correct branch: master for new features; 3.2.x, 3.1.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master / 3.2.x](https://area51.phpbb.com/docs/master/coding-guidelines.html), [3.1.x](https://area51.phpbb.com/docs/31x/coding-guidelines.html)
- [x] Commit follows commit message [format](https://wiki.phpbb.com/Git#Commit_Messages)

https://tracker.phpbb.com/browse/PHPBB3-14670

### CODE REVIEW

File: phpBB/composer.json [hunk 1]
```diff
@@ -25,41 +25,40 @@
 		"phpbb/phpbb-core": "self.version"
 	},
 	"require": {
-		"php": ">=5.4,<7.1",
+		"php": ">=5.4",
```
Comment (Crizz0) [inline phpBB/composer.json]: Symfony 3.1 need PHP 5.5.9

**Comment by marc1706** [comment]
I just pushed this to have tests running. This will need a few more fixes probably.

File: tests/di/fixtures/config/production/container/environment.yml [hunk 2]
```diff
@@ -8,7 +8,7 @@ services:
     dbal.conn:
         class: phpbb\db\driver\factory
         arguments:
-            - @service_container
+            - '@service_container'
```
Comment (lavigor) [inline tests/di/fixtures/config/production/container/environment.yml]: Leave a line note

**Comment by Nicofuma** [comment]
another BC break to take in account: https://github.com/symfony/symfony/pull/19114

**Comment by marc1706** [comment]
Ok. We should start a page of BC breaks in symfony so we don't loose track of this. I also still need to figure out what happened to the stupid UI tests.

**Comment by nicolas-grekas** [comment]
@nicofuma can you please explain how the PR you linked is affecting you?

**Comment by marc1706** [comment]
@VSEphpbb so laravel homestead prevents this from working with symfony 3.x. Same issue as with sami: it requires symfony 2.x. And homestead 3.x requires PHP 5.5.9

**Comment by marc1706** [comment]
@nicolas-grekas I guess this was just mentioned by @Nicofuma as a heads up. I can see that this was reverted. Does this also mean it has been reverted for the 3.x versions of symfony?

**Comment by VSEphpbb** [comment]
@marc1706 So can we just update the homestead dep, even though it uses PHP 5.5.9? I mean, the vagrant component is meant for local use by developers so I'm not sure it needs to be strictly limited by phpBB's PHP requirement.

**Comment by marc1706** [comment]
True, but it might be good to change the PHP version limit again.

**Comment by VSEphpbb** [comment]
Ah yes, well for 3.3 I'm pretty sure @Nicofuma wanted it bumped up to 5.5.x anyways.

**Comment by marc1706** [comment]
Removed WIP label

**Comment by marc1706** [comment]
Should probably add the PHP 7.1 builds to 3.2, too, once the s9e text formatter gets updated.

**Comment by Nicofuma** [comment]
:+1:  and we need to discuss 3.3 PHP target version

**Pull request merged**